tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
-            part_info = s3_client.initiate_upload_part(
+        resp_dir["origin_dir"] = origin_dir
+        resp_dir["results_dir"] = results_dir
+        # 1. Create a bucket using S3
+        bucket_name = c_scope_s3client.create_bucket()
+        resp_dir["bucket_name"] = bucket_name
+        # 2. Write multipart objects to the bucket
+        object_names = generate_random_files(
+            origin_dir,
+            amount,
+            min_size="20M",
+            max_size="30M",
+        )
+        resp_dir["object_names"] = object_names
+        # Upload multipart object
+        log.info("Initiate multipart upload process")
+        for i in range(len(object_names)):
+            get_upload_id = c_scope_s3client.initiate_multipart_object_upload(
                 bucket_name,
                 object_names[i],
-                part_id,
-                get_upload_id,
-                part_data[pd],
             )
-            all_part_info.append({"PartNumber": part_id, "ETag": part_info["ETag"]})
-        list_mp_uploads = s3_client.list_multipart_upload(bucket_name)
-        log.info(list_mp_uploads)
-        mp_response = s3_client.complete_multipart_object_upload(
-            bucket_name, object_names[i], get_upload_id, all_part_info
+            resp_dir[f"{object_names[i]}_upload_id"] = get_upload_id
+            all_part_info = []
+            file_name = origin_dir + "/" + object_names[i]
+            part_size = "10M"
+            log.info(f"Split data into {part_size} size")
+            part_data = split_file_data_for_multipart_upload(file_name, part_size)
+            log.info("Initiate part uploads for multipart object")
+            for pd in range(len(part_data)):
+                part_id = pd + 1
+                part_info = c_scope_s3client.initiate_upload_part(
+                    bucket_name,
+                    object_names[i],
+                    part_id,
+                    get_upload_id,
+                    part_data[pd],
+                )
+                all_part_info.append({"PartNumber": part_id, "ETag": part_info["ETag"]})
+            resp_dir["all_part_info"] = all_part_info
+        return resp_dir
+
+    def test_s3_multipart_upload(
+        self,
+        c_scope_s3client,
+        tmp_directories_factory,
+    ):
+        """
+
+        Test basic s3 operations using a noobaa bucket:
+        1. Create an account
+        2. Create a bucket using S3
+        3. Write multipart objects to the bucket
+        4. List multipart objects from the bucket
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
```

Comment:
f"{obj_name}" is equivalent to obj_name - see the other comment for suggestoins

Reply:
Fixed